tools/lsp/preview/native.rs

```diff
@@ -33,32 +33,55 @@ static GUI_EVENT_LOOP_STATE_REQUEST: LazyLock<Mutex<RequestedGuiEventLoopState>>
 
 thread_local! {static CLI_ARGS: std::cell::OnceCell<crate::Cli> = Default::default();}
 
-pub fn run_in_ui_thread<F: Future<Output = ()> + 'static>(
-    create_future: impl Send + FnOnce() -> F + 'static,
-) -> Result<(), String> {
-    // Wake up the main thread to start the event loop, if possible
-    {
-        let mut state_request = GUI_EVENT_LOOP_STATE_REQUEST.lock().unwrap();
-        if *state_request == RequestedGuiEventLoopState::Uninitialized {
-            *state_request = RequestedGuiEventLoopState::StartLoop;
-            GUI_EVENT_LOOP_NOTIFIER.notify_one();
-        }
-        // We don't want to call post_event before the loop is properly initialized
-        while *state_request == RequestedGuiEventLoopState::StartLoop {
-            state_request = GUI_EVENT_LOOP_NOTIFIER.wait(state_request).unwrap();
-        }
+fn fully_start_event_loop() -> Result<(), String> {
+    let mut state_request = GUI_EVENT_LOOP_STATE_REQUEST.lock().unwrap();
+    if *state_request == RequestedGuiEventLoopState::Uninitialized {
+        *state_request = RequestedGuiEventLoopState::StartLoop;
+        GUI_EVENT_LOOP_NOTIFIER.notify_one();
+    }
+    // We don't want to call post_event before the loop is properly initialized
+    while *state_request == RequestedGuiEventLoopState::StartLoop {
+        state_request = GUI_EVENT_LOOP_NOTIFIER.wait(state_request).unwrap();
+    }
 
-        if let RequestedGuiEventLoopState::InitializationError(err) = &*state_request {
-            return Err(err.clone());
-        }
+    if let RequestedGuiEventLoopState::InitializationError(err) = &*state_request {
+        return Err(err.clone());
     }
 
+    Ok(())
+}
+
+fn run_in_ui_thread<F: Future<Output = ()> + 'static>(
```

Comment:
Since it's only used once in lsp_to_preview_message, i think it'd be best to inline this function here. So nobody call this function without calling fully_start_event_loop yet

Reply:
I just moved the entire thing into the lsp_to_preview_message function now.